probe/endpoint/ebpf.go

```diff
@@ -123,32 +163,34 @@ func (t *EbpfTracker) run() {
 		}
 		destPort := uint16(dPort)
 
-		var evt event
-		switch line[0] {
-		case "connect":
-			evt = Connect
-		case "accept":
-			evt = Accept
-		case "close":
-			evt = Close
-		}
+		netns := line[6]
 
-		e := ConnectionEvent{
-			Type:          evt,
-			Pid:           pid,
-			SourceAddress: sourceAddr,
-			DestAddress:   destAddr,
-			SourcePort:    sourcePort,
-			DestPort:      destPort,
-		}
+		tuple := fourTuple{sourceAddr.String(), destAddr.String(), sourcePort, destPort}
 
-		t.Events = append(t.Events, e)
+		t.handleFlow(eventStr, tuple, pid, netns)
 	}
 }
 
-// WalkEvents - walk through the connectionEvents
-func (t EbpfTracker) WalkEvents(f func(ConnectionEvent)) {
-	for _, event := range t.Events {
-		f(event)
+// walkFlows calls f with all active flows and flows that have come and gone
+// since the last call to walkFlows
+func (t *EbpfTracker) walkFlows(f func(ebpfConnection)) {
+	t.Lock()
+	defer t.Unlock()
+
+	log.Infof("EbpfTracker: WalkConnections activeFlows: %d bufferedFlows: %d", len(t.activeFlows), len(t.bufferedFlows))
+
+	for _, flow := range t.activeFlows {
+		f(flow)
+	}
+	for _, flow := range t.bufferedFlows {
+		f(flow)
```

Comment:
We iterate over the closed connections exactly 1 time for 1 report, for short lived connections.